packages/playwright/src/matchers/expect.ts

```diff
@@ -129,7 +129,14 @@ function createExpect(info: ExpectMetaInfo) {
 
       if (property === 'extend') {
         return (matchers: any) => {
-          expectLibrary.extend(matchers);
+          const wrappedMatchers: any = {};
+          Object.entries(matchers).forEach(([name, matcher]) => {
+            wrappedMatchers[name] = function (...args: any[]) {
+              this.timeout = currentExpectTimeout({});
```

Comment:
As discussed offline, let's create a new `this` object here, and also update our matchers to use `this.timeout`.